packages/nc-gui/composables/useSelectedCellKeyupListener/index.ts

```diff
@@ -0,0 +1,20 @@
+import type { Ref } from 'vue'
+import { onUnmounted, useEventListener } from '#imports'
+
+export function useSelectedCellKeyupListener(selected: Ref<boolean>, handler: (e: KeyboardEvent) => void) {
+  const cleanup: Ref<ReturnType<typeof useEventListener> | null> = ref(null)
+
+  watch(selected, (value) => {
+    if (value) {
+      cleanup.value = useEventListener('keydown', handler, true)
+    } else {
+      cleanup.value?.()
+    }
+  })
+
+  onUnmounted(() => cleanup.value?.())
+
+  return {
+    cleanup,
+  }
```

Comment:
I'm not sure if `useEventListener` is the right call here, as it's binding event listeners *on mount* and unbinds *on unmount* (you can call the returned cleanup, sure, but still the bindings are timed to lifecycle hooks)

Instead just using `addEventListener` and `removeEventListener` might be cleaner code.

```ts
function useSelectedCellKeyupListener(selected: Ref<boolean>, handler: (e: KeyboardEvent) => void) {
  // prevent binding listeners on server side
  if (isClient) {
    watch(selected, (nextVal, _, cleanup) => {
      // bind listener when `selected` is truish
      if (nextVal) {
        document.addEventListener('keyup', handler)
      }

      // cleanup is called whenever the watcher is re-evaluated or stopped
      cleanup(() => {
        document.removeEventListener('keyup', handler)
      })
    })
  }
}
```

Reply:
Ya true I will update that part :+1: